cpu/sam0_common/periph/spi.c

```diff
@@ -355,35 +312,64 @@ void spi_init(spi_t bus)
     if (_is_qspi(bus)) {
         _init_qspi(bus);
     } else {
-        _init_spi(bus, dev(bus));
+        _init_spi(bus);
     }
 
     /* put device back to sleep */
     poweroff(bus);
 }
 
+static void _attach_pins(spi_t bus)
+{
+    if (gpio_is_valid(spi_config[bus].mosi_pin)) {
+        gpio_init_mux(spi_config[bus].mosi_pin, spi_config[bus].mosi_mux);
+    }
+
+    if (gpio_is_valid(spi_config[bus].miso_pin)) {
+        gpio_init_mux(spi_config[bus].miso_pin, spi_config[bus].miso_mux);
+    }
+
+    if (gpio_is_valid(spi_config[bus].clk_pin)) {
+        gpio_init_mux(spi_config[bus].clk_pin, spi_config[bus].clk_mux);
+    }
+}
+
+static void _detach_pins(spi_t bus)
+{
+    if (gpio_is_valid(spi_config[bus].mosi_pin)) {
+        gpio_disable_mux(spi_config[bus].mosi_pin);
+    }
+
+    if (gpio_is_valid(spi_config[bus].miso_pin)) {
+        gpio_disable_mux(spi_config[bus].miso_pin);
+    }
+
+    if (gpio_is_valid(spi_config[bus].clk_pin)) {
+        gpio_disable_mux(spi_config[bus].clk_pin);
+    }
+}
+
+#if MODULE_PERIPH_SPI_GPIO_MODE
 int spi_init_with_gpio_mode(spi_t bus, const spi_gpio_mode_t* mode)
```

Comment:
this function is called by `spi_init_pins()` which is caled by `spi_init` -> needs to be always defined 
I think you wanted to place that `#if`  inside the function

Reply:
Good catch, fixed